source/Calamari/Kubernetes/Commands/KubernetesApplyRawYamlCommand.cs

```diff
@@ -0,0 +1,116 @@
+using Calamari.Common.Features.Packages;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Features.StructuredVariables;
+using Calamari.Common.Features.Substitutions;
+using Calamari.Common.Plumbing.Deployment;
+using Calamari.Common.Plumbing.Deployment.Journal;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment;
+using Calamari.Deployment.Conventions;
+using Calamari.FeatureToggles;
+using Calamari.Kubernetes.Conventions;
+using Calamari.Kubernetes.ResourceStatus;
+
+namespace Calamari.Kubernetes.Commands
+{
+    [Command(Name, Description = "Apply Raw Yaml to Kubernetes Cluster")]
+    public class KubernetesApplyRawYamlCommand : Command
+    {
+        private const string Name = "kubernetes-apply-raw-yaml";
+
+        private readonly ILog log;
+        private readonly IDeploymentJournalWriter deploymentJournalWriter;
+        private readonly IVariables variables;
+        private readonly ICommandLineRunner commandLineRunner;
+        private readonly ICalamariFileSystem fileSystem;
+        private readonly IExtractPackage extractPackage;
+        private readonly ISubstituteInFiles substituteInFiles;
+        private readonly IStructuredConfigVariablesService structuredConfigVariablesService;
+        private readonly ResourceStatusReportExecutor statusReportExecutor;
+
+        private PathToPackage pathToPackage;
+
+        public KubernetesApplyRawYamlCommand(
+            ILog log,
+            IDeploymentJournalWriter deploymentJournalWriter,
+            IVariables variables,
+            ICommandLineRunner commandLineRunner,
+            ICalamariFileSystem fileSystem,
+            IExtractPackage extractPackage,
+            ISubstituteInFiles substituteInFiles,
+            IStructuredConfigVariablesService structuredConfigVariablesService,
+            ResourceStatusReportExecutor statusReportExecutor)
+        {
+            this.log = log;
+            this.deploymentJournalWriter = deploymentJournalWriter;
+            this.variables = variables;
+            this.commandLineRunner = commandLineRunner;
+            this.fileSystem = fileSystem;
+            this.extractPackage = extractPackage;
+            this.substituteInFiles = substituteInFiles;
+            this.structuredConfigVariablesService = structuredConfigVariablesService;
+            this.statusReportExecutor = statusReportExecutor;
+            Options.Add("package=", "Path to the NuGet package to install.", v => pathToPackage = new PathToPackage(Path.GetFullPath(v)));
+        }
+        public override int Execute(string[] commandLineArguments)
+        {
+            if (!FeatureToggle.GitSourcedYamlManifestsFeatureToggle.IsEnabled(variables))
+                throw new InvalidOperationException(
+                    "Unable to execute Kubernetes Apply Raw Yaml Command because the appropriate feature has not been enabled.");
```

Comment:
Even though this command should only be executed when the feature toggle is switched on, if it somehow executes then it will fail with this error message.

Reply:
Good suggestion 👍